File: the-lambda-trilogy/python/lambdas/the_fat_lambda/fatlambda.py
def extract_params(event):
    try:
        first_num = event["queryStringParameters"]['firstNum']
    except KeyError:
        first_num = 0

    try:
        second_num = event["queryStringParameters"]['secondNum']
    except KeyError:
        second_num = 0

    return first_num, second_num


def add(event, context):
    first_num, second_num = extract_params(event)

    result = int(first_num) + int(second_num)
    print("The result of % s + % s = %s" % (first_num, second_num, result))
    return {'body': result, 'statusCode': 200}


def subtract(event, context):
    first_num, second_num = extract_params(event)

    result = int(first_num) - int(second_num)
    print("The result of % s - % s = %s" % (first_num, second_num, result))
    return {'body': result, 'statusCode': 200}


def multiply(event, context):
    first_num, second_num = extract_params(event)

    result = int(first_num) * int(second_num)
    print("The result of % s * % s = %s" % (first_num, second_num, result))
    return {'body': result, 'statusCode': 200}

File: the-lambda-trilogy/typescript/lambda-fns/fat-lambda/fat-lambda.ts
export {};

/**
 * This lambda contains 3 handlers in the same file rather than separating them out
 * 
 * Note that crucially this still ends up 3 lambdas they just have all the code for all 3
 * in each one
 */

exports.add = async (event:any) => {
    console.log(JSON.stringify(event, null, 2));

    // pull firstNum and secondNum from queryparams, default to 0
    let firstNum = event?.queryStringParameters?.firstNum ?? 0;
    let secondNum = event?.queryStringParameters?.secondNum ?? 0;

    let result = Number(firstNum) + Number(secondNum);
    console.log(`result of ${firstNum} + ${secondNum} = ${result}`)

    return sendRes(200, result.toString());
}

exports.subtract = async (event:any) => {
    console.log(JSON.stringify(event, null, 2));

    // pull firstNum and secondNum from queryparams, default to 0
    let firstNum = event?.queryStringParameters?.firstNum ?? 0;
    let secondNum = event?.queryStringParameters?.secondNum ?? 0;

    let result = firstNum - secondNum;
    console.log(`result of ${firstNum} - ${secondNum} = ${result}`)

    return sendRes(200, result.toString());
}

exports.multiply = async (event:any) => {
    console.log(JSON.stringify(event, null, 2));

    // pull firstNum and secondNum from queryparams, default to 0
    let firstNum = event?.queryStringParameters?.firstNum ?? 0;
    let secondNum = event?.queryStringParameters?.secondNum ?? 0;

    let result = firstNum * secondNum;
    console.log(`result of ${firstNum} x ${secondNum} = ${result}`)

    return sendRes(200, result.toString());
}

const sendRes = (status:number, body:string) => {
    var response = {
      statusCode: status,
      headers: {
        "Content-Type": "text/html"
      },
      body: body
    };
    return response;
  };